sdk/appconfiguration/app-configuration/src/internal/queryParamPolicy.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type {
+  PipelinePolicy,
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest,
+} from "@azure/core-rest-pipeline";
+
+/**
+ * Creates a PipelinePolicy that normalizes query parameters:
+ *  - Lowercase names
+ *  - Sort by lowercase name
+ *  - Preserve the relative order of duplicates
+ */
+export function queryParamPolicy(): PipelinePolicy {
+  return {
+    name: "queryParamPolicy",
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      try {
+        const originalUrl: string = request.url;
+        const url = new URL(originalUrl);
+
+        if (url.search === "") {
+          return next(request);
+        }
+
+        const params: ParamEntry[] = [];
+        for (const [name, value] of url.searchParams.entries()) {
+          params.push({ lowercaseName: name.toLowerCase(), value });
```

Comment:
This approach has problems that we didn't discuss. At least, I think. Perhaps you can confirm.

Consider if I have two keys in App Configuration.

`%2F-one`
`%2F-two`

Then I do `client.listConfigurationSettings(keyFilter: "%2F*")`.

Would this break? My expectation is the decoding that `URLSearchParams` is doing will break this.

We may need to just split on '&'